foreman/data_refinery_foreman/foreman/main.py

```diff
@@ -391,60 +385,76 @@ def requeue_downloader_job(last_job: DownloaderJob) -> bool:
         else:
             # Can't communicate with nomad just now, leave the job for a later loop.
             new_job.delete()
-            return False
+            return False, ""
     except:
         logger.error("Failed to requeue Downloader Job which had ID %d with a new Downloader Job with ID %d.",
                      last_job.id,
                      new_job.id)
         # Can't communicate with nomad just now, leave the job for a later loop.
         new_job.delete()
-        return False
+        return False, ""
 
-    return True
+    return True, new_job.volume_index
 
 
-def count_downloader_jobs_in_queue(nomad_client: Nomad) -> int:
+def count_downloader_jobs_in_queue(window=datetime.timedelta(minutes=5)) -> int:
     """Counts how many downloader jobs in the Nomad queue do not have status of 'dead'."""
-    try:
-        all_downloader_jobs = nomad_client.jobs.get_jobs(prefix="DOWNLOADER")
-    except:
-        # Nomad is down, return an impossibly high number to prevent
-        # additonal queuing from happening:
-        return sys.maxsize
 
-    total = 0
-    for job in all_downloader_jobs:
-        if job['ParameterizedJob'] and job['JobSummary'].get('Children', None):
-            total = total + job['JobSummary']['Children']['Pending']
-            total = total + job['JobSummary']['Children']['Running']
+    nomad_host = get_env_variable("NOMAD_HOST")
+    nomad_port = get_env_variable("NOMAD_PORT", "4646")
+    nomad_client = Nomad(nomad_host, port=int(nomad_port), timeout=30)
+
+    global TIME_OF_LAST_DOWNLOADER_JOB_CHECK
+    global DOWNLOADER_JOBS_IN_QUEUE
+
+    if (timezone.now() - TIME_OF_LAST_DOWNLOADER_JOB_CHECK > window):
+        TIME_OF_LAST_DOWNLOADER_JOB_CHECK = timezone.now()
+        try:
+            all_downloader_jobs = nomad_client.jobs.get_jobs(prefix="DOWNLOADER")
+        except:
+            # Nomad is down, return an impossibly high number to prevent
+            # additonal queuing from happening:
+            DOWNLOADER_JOBS_IN_QUEUE = sys.maxsize
+
+        total = 0
+        for job in all_downloader_jobs:
```

Comment:
If nomad is down `all_downloader_jobs` will be undefined. I think this block should probably be inside the `try`.